python/operations.cpp

```diff
@@ -2,34 +2,280 @@
+    Coords, and masks for any but the given dimension are required to match and are copied to the output without changes.
+    In the case of a dataset, the output contains only items that are present in both inputs.)",
+      // Raises
+      "If the dtype or unit does not match, or if the dimensions and shapes "
+      "are incompatible.",
+      // See also
+      ":py:class:`scipp.sum`",
+      // Returns
+      "New variable, data array, or dataset containing the concatenated data.",
+      // Return type
+      // "Variable, DataArray, or Dataset.",
+      // Input parameters
+      {{"x", "First Variable, DataArray, or Dataset."},
+       {"y", "Second Variable, DataArray, or Dataset."},
+       {"dim", "Dimension over which to concatenate."}}};
+  bind_free_function<Variable, VAConstView, VAConstView, const Dim>(
+      concatenate, "concatenate", m, docs);
+  bind_free_function<DataArray, DAConstView, DAConstView, const Dim>(
+      concatenate, "concatenate", m, docs);
+  bind_free_function<Dataset, DSConstView, DSConstView, const Dim>(
+      concatenate, "concatenate", m, docs);
+
+  // Dot-product
+  docs = {"Element-wise dot-product.",
+          "If the dtype is not a vector such as "
+          ":py:class:`scipp.dtype.vector_3_double.`",
+          "",
+          "Variable, data array, or dataset with scalar elements based on the "
+          "two inputs.",
+          // "Variable, DataArray, or Dataset.",
+          {{"x", "Left operand Variable, DataArray, or Dataset."},
+           {"y", "Right operand Variable, DataArray, or Dataset."}}};
+  bind_free_function<Variable, VAConstView, VAConstView>(dot, "dot", m, docs);
+
+  // Sort
+  docs = {"Sort variable along a dimension by a sort key.",
+          "If the key is invalid, e.g., if it has not exactly one dimension, "
+          "or if its dtype is not sortable.",
+          "",
+          "Sorted variable, data array, or dataset.",
+          // "Variable, DataArray, or Dataset.",
+          {{"x", "Variable, DataArray, or Dataset to be sorted."},
+           {"key", "Variable, DataArray, or Dataset as sorting key."}}};
+  bind_free_function<Variable, VAConstView, VAConstView>(sort, "sort", m, docs);
+  bind_free_function<DataArray, DAConstView, VAConstView>(sort, "sort", m,
+                                                          docs);
+  bind_free_function<Dataset, DSConstView, VAConstView>(sort, "sort", m, docs);
+  docs.set_description("Sort data array along a dimension by the coordinate "
+                       "values for that dimension.");
+  docs.set_raises("If the key is invalid, e.g., if it is not a Dim.");
+  docs.set_param(1, {"dim", "Dimension over which to sort."});
+  bind_free_function<DataArray, DAConstView, const Dim &>(sort, "sort", m,
+                                                          docs);
+  bind_free_function<Dataset, DSConstView, const Dim &>(sort, "sort", m, docs);
+
+  // Absolute value
+  docs = {"Element-wise absolute value.",
```

Comment:
I liked Simon's kwarg solutions around `makeVariable`. Maybe could be used here? Maybe too heavyweight?

Reply:
`kwargs` would be much better here, yes. Thanks for the idea.

Reply:
Myabe something less heavy could be a syntax like
```
Docstring().description("Element-wise absolute value.").raises("If input is integer")...
```
and the methods would set the right members in succession?
I guess a bit like the `pybind11` `.def(...).def(...)` ?

Reply:
So you lose the standardization of all docstrings, but it easier to make changes to Docstring in the future without having to change all bindings, as Simon points out below.

Reply:
I guess what you are suggesting is much like the `builder` pattern. You would then not need to call your named methods in a particular order.